bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/extensions/ExtensionResource.java

```diff
@@ -149,6 +153,25 @@ public void run() {
     }
 
     @POST
+    @Path("/url/{url}/install")
+    @ApiOperation(value = "Installs the extension from the given URL.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 400, message = "The given URL is malformed or not valid.") })
+    public Response installExtensionByURL(
+            final @PathParam("url") @ApiParam(value = "extension install URL", required = true) String url) {
+        try {
+            URI extensionURI = new URI(url);
+            String extensionId = getExtensionId(extensionURI);
+            installExtension(extensionId);
+        } catch (Exception e) {
```

Comment:
So, why catching all exceptions and not the one (`URISyntaxException`) that is thrown (IMHO it is a little bit a difference between checked and unchecked ones, if you want to ensure the caller does never see a strack trace, you could catch all uncecked one, but checked one should be catched as needed only)?

There are exceptions that needs to be handled differently (e.g. InterruptedException) in the most cases. If you catch all exceptions this part of code never gets identified by the IDE to be changed if in the call tree other checked exceptions are added.

I assume this is not only a personal point of view but a "common quality topic", see e.g. http://checkstyle.sourceforge.net/config_coding.html#IllegalCatch